samples/TagHelperSample.Web/Components/FeaturedMoviesComponent.cs

```diff
@@ -11,31 +13,38 @@ namespace TagHelperSample.Web.Components
     [ViewComponent(Name = "FeaturedMovies")]
     public class FeaturedMoviesComponent : ViewComponent
     {
-        private MoviesService _moviesService;
+        private readonly IMemoryCache _cache;
+        private readonly MoviesService _moviesService;
 
-        public FeaturedMoviesComponent(MoviesService moviesService)
+        public FeaturedMoviesComponent(MoviesService moviesService, IMemoryCache cache)
         {
             _moviesService = moviesService;
+            _cache = cache;
         }
 
         public IViewComponentResult Invoke()
         {
-            IExpirationTrigger trigger;
-            var movies = _moviesService.GetFeaturedMovies(out trigger);
-            
-            // Add custom triggers
-            EntryLinkHelpers.ContextLink.AddExpirationTriggers(new[] { trigger });
+            var cacheKey = "featured_movies";
+            IEnumerable<FeaturedMovies> movies;
+            if (!_cache.TryGetValue(cacheKey, out movies))
+            {
+                IExpirationTrigger trigger;
+                movies = _moviesService.GetFeaturedMovies(out trigger);
+                _cache.Set(cacheKey, movies, new MemoryCacheEntryOptions().AddExpirationTrigger(trigger));
```

Comment:
Maybe make a comment this also affects the caching of the base thing. It's not particularly obvious looking at this.